source/addins/ArcMapAddinDistanceAndDirection/ArcMapAddinDistanceAndDirection/ViewModels/TabBaseViewModel.cs

```diff
@@ -941,6 +941,68 @@ internal ILayer GetLayerFromMapByName(IMap map, string name)
             return null;
         }
 
+        /// <summary>
+        /// Returns the extent of a basemap
+        /// </summary>
+        /// <param name="map">IMap interface</param>
+        /// <returns>IEnvelope interface</returns>
+        internal IEnvelope GetBasemapExtent(IMap map)
+        {
+            try
+            {
+                ILayer basemapLayer = null;
+                var layers = map.get_Layers();
+                var layer = layers.Next();
+                while (layer != null)
+                {
+                    if (layer is ICompositeLayer)
```

Comment:
I'm not sure this is a full-proof way of detecting the BaseMapLayer:

1. many different types of layers implement this interface (so just the first group layer would be returned)
2. user may be using just a simple layer/feature class as basemap (ex. world/continents shapefile)

Would it be possible just to use ActiveView.FullExtent (I think that is what is used when you hit the zoom to full extent button)?

Reply:
@csmoore I'll test with full extent